test/e2e/alerting_rule_test.go

```diff
@@ -0,0 +1,277 @@
+// Copyright 2023 The Cluster Monitoring Operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"strings"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	statusv1 "github.com/openshift/api/config/v1"
+	osmv1 "github.com/openshift/api/monitoring/v1"
+	monv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+
+	"github.com/pkg/errors"
+
+	"github.com/openshift/cluster-monitoring-operator/test/e2e/framework"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	validArName   = "valid-test-alerting-rule"
+	invalidArName = "invalid-test-alerting-rule"
+)
+
+func TestAlertingRule(t *testing.T) {
+
+	ctx := context.Background()
+	alertingRules := f.OpenShiftMonitoringClient.MonitoringV1().AlertingRules(f.Ns)
+
+	ar := &osmv1.AlertingRule{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      invalidArName,
+			Namespace: f.Ns,
+			Labels: map[string]string{
+				framework.E2eTestLabelName: framework.E2eTestLabelValue,
+			},
+		},
+		Spec: osmv1.AlertingRuleSpec{
+			Groups: []osmv1.RuleGroup{
+				{
+					Name:     "test-group",
+					Interval: "15s",
+					Rules: []osmv1.Rule{
+						{
+							Alert:       "InvalidAlert",
+							Expr:        intstr.FromString("invalid_expr("),
+							For:         "5m",
+							Labels:      map[string]string{"severity": "high"},
+							Annotations: map[string]string{"summary": "high foo"},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// The operator is healthy
+	f.AssertOperatorCondition(statusv1.OperatorDegraded, statusv1.ConditionFalse)
```

Comment:
it'd be great to have this check before each test case (that assumes the operator to be healthy), it'd make debugging easier.